syntax = "proto2";

package apollo.planning;

message NaviPathDeciderConfig {
  optional double min_path_length = 1 [default = 5];
  optional uint32 min_look_forward_time = 2 [default = 2];
  optional double max_keep_lane_distance = 3 [default = 0.8];
  optional double max_keep_lane_shift_y = 4 [default = 20.0];
  optional double min_keep_lane_offset = 5 [default = 15.0];
  optional double keep_lane_shift_compensation = 6 [default = 0.01];
  optional MoveDestLaneConfigTable move_dest_lane_config_talbe = 7;
  optional double move_dest_lane_compensation = 8 [default = 0.35];
  optional double max_kappa_threshold = 9 [default = 0.0];
  optional double move_dest_lane_compensation_by_kappa = 10 [default = 0.0];
  optional bool enable_plan_start_point = 11 [default = false];
  optional double basic_path_lateral_compensation = 12 [default = 0.0];
  optional double max_nudge_distance = 13 [default = 0.10];
  optional double nudge_compensation = 14 [default = 0.001];
  optional bool enable_quintic_polynomial_curve_path = 15 [default = false];
  optional double move_dest_lane_longitudinal_coef = 16 [default = 20.0];
  optional double move_dest_lane_velocity_coef = 17 [default = 0.10];
  optional double move_dest_lane_kappa_coef = 18 [default = 20.0];
  optional double move_dest_lane_longitudinal_buffer = 19 [default = 0.8];
  optional double dense_sample_resolution = 20 [default = 0.5];
  optional double sparse_sample_resolution = 21 [default = 2.0];
  optional double move_curve_dl_velocity_coef = 22 [default = 0.01];
  optional double move_to_left_curve_start_dl = 23 [default = 0.008];
  optional double move_to_right_curve_start_dl = 24 [default = -0.015];
  optional double min_large_curvature = 25 [default = 0.015];
}

message MoveDestLaneConfigTable {
  repeated ShiftConfig lateral_shift = 1;
}

message ShiftConfig {
  optional double max_speed = 1 [default = 4.16];
  optional double max_move_dest_lane_shift_y = 3 [default = 0.4];
}
